### PR TITLE
Generate @export instead of externs for exporting decorators.

### PR DESCRIPTION
**Pull request by drjanitor**
Avoids compiler explosions when multiple files export symbols with the same name. Also, the closure compiler documentation really, really wants you to use exports instead of externs whenever possible.

### CODE REVIEW

**Comment by drjanitor** [comment]
Done, and done.

**Comment by evmar** [comment]
I guess this talks about this issue:
https://developers.google.com/closure/compiler/docs/api-tutorial3#no

File: test_files/exporting_decorator/exporting.ts [hunk 1]
```diff
@@ -1,50 +1,47 @@
...
+function exportingDecorator() {
   return function(target: any, name: string|symbol) {}
 }
 
-function nonExportingFieldDecorator() {
+function nonExportingDecorator() {
   return function(target: any, name: string|symbol) {}
 }
 
-@exportingClassDecorator()
```
Comment (evmar) [inline test_files/exporting_decorator/exporting.ts]: Why did you remove this?  Is it related?

**Comment by drjanitor** [comment]
turns out you can't @export classes inside a goog.module, so I removed that
from tests.

**Comment by evmar** [comment]
Does that mean you can't use this for classes?  Doesn't that break the use case?

**Comment by drjanitor** [comment]
the use case I'm after is "do not remove this field from the class", which
is indeed working. I don't think I'm actually afraid of anyone
instantiating my class from a template or something else.

**Comment by drjanitor** [comment]
Ping?

**Comment by evmar** [comment]
Oops, sorry, meant to approve before.  I am a bit worried that this feature is very particular to your use case and not generally reusable by other users.
